### ISSUE TITLE
React: Zoom window updates

### ISSUE THREAD

**Issue by TimVanMourik** [TimVanMourik/GiraffeTools#78]
The current canvas for porcupine node editor is a `PinchView` from [`react-pinch-zoom-pan`](https://github.com/gerhardsletten/react-pinch-zoom-pan, awesome work @gerhardsletten!). I would like to modify the content of this window for our purposes.

* [ ] I would like to have the Canvas auto-expand when a node is dragged outside the current frame.
* [ ] Although the zoom rendering is incredibly smooth, the alt-click zoom is very glitchy. Either fix this or find a reasonable alternative.
* [ ] Allow zooming by means of + and - button in the corner of the screen

![untitled](https://user-images.githubusercontent.com/6152205/42760326-6644d3e0-890a-11e8-896b-749d751db3c4.png)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.07 ETH (33.19 USD @ $474.21/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/TimVanMourik/GiraffeTools/78/801).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $27,277.24 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by TimVanMourik** [comment]
@owocki, @PixelantDesign this bounty's got [funding](https://gitcoin.co/issue/TimVanMourik/GiraffeTools/78/801) and an approved worker @hubris but no @gitcoinbot message. This happened earlier and a Gitcoin issues was made in https://github.com/gitcoinco/web/issues/1627

**Comment by TimVanMourik** [comment]
Hi @hubris! I approved your application, although gitcoinbot is quiet today. Could you please drop a message, either here or on Slack? 🙂

**Comment by PixelantDesign** [comment]
Thanks for the heads up @TimVanMourik, I had this problem last night too. We're looking into this.

**Comment by owocki** [comment]
@TimVanMourik Github suspended Gitcoinbot and thats why there is no message here.  We just emailed them and quickly got it unsuspended.  Thanks for the heads up!

**Comment by hubris** [comment]
@TimVanMourik Hi, I will try to start this evening (GMT+1). In any case I will be fully on it friday.

**Comment by TimVanMourik** [comment]
Thanks @owocki, @PixelantDesign for the heads-up!

And you too @hubris, let me know if you've got any questions, either here or via Slack!

**Comment by hubris** [comment]
@TimVanMourik On which repo/branch should I start? Thank you

**Comment by TimVanMourik** [comment]
Ah, sorry, `dev` branch, I should have mentioned! 
Yes, I know the zoom is locked to ‘=‘. The inconvenient thing about plus is that it’s behind a shift key. What do you think works best in terms of UX?

**Comment by hubris** [comment]
@TimVanMourik Since react-pinch-zoom-pan is not a lot of code maybe the most straightforward solution could be to integrate the 2 files in the code base and do all the needed changes from here.

Desktop zoom could use the wheel event for zooming and right click for panning.

**Comment by TimVanMourik** [comment]
Yes, I think I agree! Would you be interested in making a shot at that for this bounty or shall I make a headstart and put it on whatever I can’t finish?

**Comment by hubris** [comment]
Considering the amount of work, the hourly rate is too low for me. If you increase the bounty I could give it a try otherwise I would leave it to somebody else. Sorry.

**Comment by hubris** [comment]
@TimVanMourik Hi, let me know what you want to do. Thank you.

**Comment by TimVanMourik** [comment]
Hi @hubris! Sorry, busy weekend! Yes, the bounty on this one is way too low for an adapted re-implementation of react-pinch-zoom-pan, sure! I happily tripple this bounty (0.21 ETH) if we can make that work with the functionality of this original issue!

**Comment by hubris** [comment]
@TimVanMourik Ok I will try to work on it this week.

**Comment by TimVanMourik** [comment]
I just found this project and it looks like exactly what we need. I'm looking into it further.

**Comment by hubris** [comment]
@TimVanMourik Hi, yes it does seem to implement most of the features needed for zoom/pan/zoom-to-fit. At first sight, it seems to need some work on the node rendering part though. I was working on the mouse wheel zoom with react-pinch-zoom-pan so I stop since it may be a better alternative. Let me know your plan on this issue. Thank you.

**Comment by TimVanMourik** [comment]
Sorry, I didn’t want to throw in a change-of-plans entirely! I merely meant it as a good example that we miht take ideas from! Either solution, or a hybrid, or anything else that works is fine 🙂 Sorry if I caused any confusion

**Comment by TimVanMourik** [comment]
Thanks! Could you please make a WIP PR for this? Makes it easier for me to look at it and make changes if necessary!

**Comment by TimVanMourik** [comment]
Thanks @hubris! I didn't have much time to look at it yet, in my weekend, but updated some things here and there to make it fit with the current Redux setup, see commit 4a5b84cba79a8781cf124ea501a7c19dae10e9d9. 

Quick remarks: Overall I like the zoom and drag functionality! It's working for me on Chrome as well. But it is a bit glitchy and slow it seems, not sure why. I noticed that some parts have jquery under the hood. That might have to do with it and if those variables could instead be passed down via state/props, I think that would be great. Also keeps the app a lot more consistent. 

I'll have a better look at it later this week, alright?

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 week, 3 days ago.
Please review their action plans below:


1. [hubris](https://gitcoin.co/profile/hubris) has been approved to start work. 
    
    Hi,
I would like to try this one. I'm a noob at react but this issue seems to be a good starting point to learn things. Let me know if you're ok with that.
Thank you

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/TimVanMourik/GiraffeTools/78/801).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 0.07 ETH  (25.03 USD @ $357.51/ETH) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $23,509.36 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by TimVanMourik** [comment]
I had a bit busier week than I thought I would, but finally got round to it. I made some sizeable edits in commits until b87017b153b0380dd2c0b93af1a468d8df8e5f79. I removed all jquery from the code and restructured here and there. The only thing that still needs to be solved is that the window currently only responds to scroll and drag when the scaled translated version of the `panview` div is hovered over. It would be nice if it worked on anywhere in the `mainCanvas`.
